internal/meta/req.go

```diff
@@ -35,6 +37,9 @@ func FromTransportRequest(ctx context.Context, req *transport.Request) yarpc.Req
 // ToTransportRequest fills the given transport request with information from
 // the given ReqMeta. The Context associated with the ReqMeta is returned.
 func ToTransportRequest(reqMeta yarpc.CallReqMeta, req *transport.Request) context.Context {
+	if reqMeta == nil {
+		log.Panic("reqMeta must not be nil")
```

Comment:
Note: before it would panic on a nil pointer error, this only adds a better error message to the panic at the end of the day.

I would argue that this is a programming error to leave a nil reqMeta and this should Panic. If people want, they can always recover the panic like for any other panic.